boa/src/symbol/mod.rs

```diff
@@ -42,65 +55,78 @@ pub struct WellKnownSymbols {
     unscopables: RcSymbol,
 }
 
+/// Reserved number of symbols.
+///
+/// This is where the well known symbol live
+/// and internal engine symbols.
+const RESERVED_SYMBOL_HASHES: u64 = 128;
+
+thread_local! {
```

Comment:
@HalidOdat what was the reason you went with thread_local! for holding these values? Nothing wrong with it just curious.

Reply:
I ask because aren’t they supposed to be the same symbols across worker threads, web workers etc?

Reply:
> @HalidOdat what was the reason you went with thread_local! for holding these values? Nothing wrong with it just curious.

Well we can't pass symbols to other threads since `Symbol` is not thread safe (contains `RcString`, and store symbols in an `RcSymbol`). So I stored in a `thread_local`. We could store it in `Arc`.
But for now I think this is a good solution to detach from `Context`, we will see what we can do later.

Reply:
Yup agree